### PR TITLE
Rate limit JWKS endpoint

### PR DESCRIPTION
**Pull request by VirajSalaka**
### Purpose
With this PR, JWKS endpoint calls are rate limited based on local rate limit feature.

### Issues

Fixes #

### Automation tests
 - Unit tests added: No
 - Integration tests added: Yes

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [ ] Assigned 'Type' label
- [ ] Assigned the project
- [ ] Validated respective github issues
- [ ] Assigned milestone to the github issue(s)

### CODE REVIEW

File: adapter/internal/oasparser/envoyconf/http_filters.go [hunk 1]
```diff
@@ -158,6 +161,24 @@ func getLuaFilter() *hcmv3.HttpFilter {
...
 }
 
+// getHTTPLocalRateLimitFilter returns the local rate limit filter which is used for JWKS endpoint specifically.
+func getHTTPLocalRateLimitFilter() *hcmv3.HttpFilter {
+	localRateLimitConfig := &local_ratelimit_v3.LocalRateLimit{
+		StatPrefix: localRateLimitStatPrefix,
+	}
+	marshalledRateLimitConfig, err := ptypes.MarshalAny(localRateLimitConfig)
+	if err != nil {
+		logger.LoggerOasparser.Error("Error while generating the local rate limit filter.", err)
```
Comment (slahirucd7) [inline adapter/internal/oasparser/envoyconf/http_filters.go:171]: Can we change this to `errorf` ?

File: integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/standalone/jwtGenerator/JwtGeneratorTestCase.java [hunk 2]
```diff
@@ -168,4 +170,18 @@ public void testJWTVerification() throws Exception {
...
         Assert.assertTrue(jwsObject.verify(verifier),"JWT failed to validate with JWKS response");
     }
+
+    @Test(description = "Test Rate limiting on JWKS", dependsOnMethods = "testJWTVerification")
+    public void testJWKSEndpointRatelimit() throws MalformedURLException, CCTestException {
+        boolean received429 = false;
+        for (int i = 0; i < 10; i++) {
+            HttpResponse res = HttpsClientRequest.doGet(Utils.getServiceURLHttps("/.wellknown/jwks"),
+                    new HashMap<>());
+            if (res.getResponseCode() == 429) {
```
Comment (renuka-fernando) [inline integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/standalone/jwtGenerator/JwtGeneratorTestCase.java:180]: Don't we need to check that, `i >= 5`

File: adapter/config/default_config.go [hunk 3]
```diff
@@ -197,6 +197,8 @@ var defaultConfig = &Config{
 					UseForSigning:         true,
 				},
 			},
+			JwksRatelimitQuota:               1000,
+			JwksRatelimitTimeWindowInSeconds: 10,
```
Comment (renuka-fernando) [inline adapter/config/default_config.go:201]: Shall we include these in, config TOML template file

File: adapter/internal/oasparser/envoyconf/http_filters.go (hunk 1 above)
Comment (VirajSalaka) [reply adapter/internal/oasparser/envoyconf/http_filters.go:171]: Errorf is only required if um going to format the error. But in this case, it is pretty much like right after the error message, we are printing the error

File: integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/standalone/jwtGenerator/JwtGeneratorTestCase.java (hunk 2 above)
Comment (VirajSalaka) [reply integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/standalone/jwtGenerator/JwtGeneratorTestCase.java:180]: i=1 is already tested (from dependsOn method)

**Comment by renuka-fernando** [comment]
There are multiple commits with the message "bug fix" Shall we squash them?

**Comment by VirajSalaka** [comment]
@renuka-fernando We can squash them just before the merge in the PR itself.

**Review by slahirucd7 — APPROVED** [review]

**Review by NomadXD — APPROVED** [review]
LGTM

File: adapter/config/default_config.go (hunk 3 above)
Comment (VirajSalaka) [reply adapter/config/default_config.go:201]: fixed

**Comment by choreo-cicd** [comment]
[failed] Controlplane cluster : dev-deployment-v2 : 20221221.5

**Comment by choreo-cicd** [comment]
[failed]  : dev-deployment-v2 : 20221221.5

**Comment by choreo-cicd** [comment]
[failed] DataplaneE1 cluster : dev-deployment-v2 : 20221221.5

**Comment by choreo-cicd** [comment]
[failed] DataplaneE2 cluster : dev-deployment-v2 : 20221221.5

**Comment by choreo-cicd** [comment]
[failed] DataplaneE2 cluster : dev-deployment-v2 : 20221221.5

**Comment by choreo-cicd** [comment]
[failed] Controlplane cluster : dev-deployment-v2 : 20221221.5

**Comment by choreo-cicd** [comment]
[failed]  : dev-deployment-v2 : 20221221.5

**Comment by choreo-cicd** [comment]
[failed] DataplaneE1 cluster : dev-deployment-v2 : 20221221.5

**Comment by choreo-cicd** [comment]
[failed] DataplaneE1 cluster : stage-deployment-v2 : 20221221.2

**Comment by choreo-cicd** [comment]
[failed] Controlplane cluster : stage-deployment-v2 : 20221221.3

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : stage-deployment-v2 : 20221221.4

**Comment by choreo-cicd** [comment]
[succeeded] DataplaneE1 cluster : stage-deployment-v2 : 20221221.4

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : prod-deployment-v2 : 20230109.2

**Comment by choreo-cicd** [comment]
[succeeded] DataplaneE1 cluster : prod-deployment-v2 : 20230109.2

**Pull request merged**